### PR TITLE
Topicsscrenn

### PR DESCRIPTION
**Pull request by hanelmouta**
This pull request introduces the "Topic Creation" screen, enabling users to add new topics directly from the Group Home screen. It includes backend integrations with the TopicViewModel for handling topic creation logic and updates to navigation routes to ensure a smooth user experience.
The changes encompass several key areas including the user interface, navigation logic, and integration with the TopicViewModel for backend operations.
I have implemented the TopicCreation Composable function, which features a text field for users to enter the name of a new topic and a button that commits the topic to the database via the TopicViewModel. This is designed to ensure that users can not only propose new topics but also see them reflected in real-time across user sessions.
Navigation has been enhanced by adding a new route, Route.TOPICCREATION, to our navigation graph. 
This route is accessed via a new button on the Group Home screen, ensuring users can navigate seamlessly to the Topic Creation screen without interrupting their workflow. 
I conducted thorough manual testing to confirm that the new screen functions as expected.

### CODE REVIEW

**Review by cemuelle — COMMENTED** [review]
Don't forget to run KTFmt.
Otherwise it seem's good except the string that you need to extract.

File: app/src/main/java/com/github/se/studybuddies/ui/groups/GroupScreen.kt [hunk 1]
```diff
@@ -84,6 +88,18 @@ fun GroupScreen(
...
       },
+      floatingActionButton = {
+          FloatingActionButton(
+              onClick = { navigationActions.navigateTo(Route.TOPICCREATION) },
+
+          ) {
+              Icon(
+                  imageVector = Icons.Default.Add,
+                  tint = White,
+                  contentDescription = "Create Topic"
```
[the remark this answers is not in the document]
Comment (hanelmouta) [reply app/src/main/java/com/github/se/studybuddies/ui/groups/GroupScreen.kt]: done

File: build.gradle.kts [hunk 2]
```diff
@@ -5,7 +5,7 @@ buildscript {
 }
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 plugins {
-    id("com.android.application") version "8.3.2" apply false
+    id("com.android.application") version "8.4.0" apply false
```
Comment (oliver216) [inline build.gradle.kts:8]: This should not be changed, why do you needed to change this ?

**Review by Ovoland — APPROVED** [review]
Overall great job, the topic creation feature is yorking properly, some UI reformating will be necessary in the future
Also beware of changes to the gradle version (should be 8.3.2 instead of 8.4.0) 

Note : this review was done by Diego Tavares on Oceane's machine

**Pull request merged**